### PR TITLE
Fixed bug in Polynomial derivative

### PR DESCRIPTION
**Pull request by CodeMaxx**
Earlier `diff()` gave derivative even w.r.t. `""`(empty symbol)
So 
```cpp
Symbol none = symbol("");
RCP<const UnivariateIntPolynomial> b = univariate_int_polynomial(none, {{0, 1_z}});
REQUIRE(b->diff(none)->__str__() == "0"); // Returned false
```
gave error.

Also for 
```cpp
b = univariate_int_polynomial(x, {{0, 1_z}, {1, 2_z}});
```
`b->diff(none)` gave `0`
 
@irislq @isuruf @chenchfort @Sumith1896

### CODE REVIEW

**Comment by irislq** [comment]
Nice, didn't think about this situation before. +1 from me.

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -437,6 +437,9 @@ static RCP<const Basic> diff(const CLASS &self, \
 
     static RCP<const Basic> diff(const UnivariateIntPolynomial &self, 
             const RCP<const Symbol> &x) {
+        if(x->__eq__(*symbol("")))
+            throw:: std::runtime_error("Derivative can't be with respect to nothing");
```
Comment (isuruf) [inline symengine/derivative.cpp]: `throw::` -> `throw`

Comment (CodeMaxx) [inline symengine/derivative.cpp]: I wonder why it didn't give a compiler error

**Comment by irislq** [comment]
@isuruf Looks like one of the Travis builds terminated prematurely. Can someone restart Travis?

**Comment by Sumith1896** [comment]
@irislq Done

**Comment by CodeMaxx** [comment]
@isuruf Ping

**Comment by irislq** [comment]
@CodeMaxx Can you solve the merge conflicts?

**Comment by isuruf** [comment]
This issue is not unique to polynomial derivative. Checking that the symbol is not equal to "" is an extra overhead in each call to `diff` which is not really needed.

**Comment by isuruf** [comment]
What was the result of,
```
Symbol none = symbol("");
RCP<const UnivariateIntPolynomial> b = univariate_int_polynomial(none, {{0, 1_z}});
b->diff(none)->__str__()
```
before?

**Comment by CodeMaxx** [comment]
@isuruf Earlier there was an error since there was an error in `is_canonical()` which was fixed by the other PR.

**Comment by isuruf** [comment]
Ah okay. See my comment here, https://github.com/symengine/symengine/pull/894#issuecomment-204941082

**Comment by CodeMaxx** [comment]
Are you sure its not required? I mean the user is not likely to do it directly but in complex computations where variables are passed around(Polynomials are used in all kinds of computations), this situation may arise due to some error by the user... The user will be able to detect this more easily if we raise an exception.

**Pull request closed without merging**